app/src/main/java/ar/com/wolox/android/example/ui/login/LoginPresenter.java

```diff
@@ -18,17 +19,26 @@
     }
 
     public void setPreferences(Context context, String emailField) {
-        sharedPref = context.getSharedPreferences(context.getString(R.string.vlogin_preferences_name), Context.MODE_PRIVATE);
+        sharedPref = context.getSharedPreferences(context.getString(R.string.login_preferences_name), Context.MODE_PRIVATE);
         editor = sharedPref.edit();
         editor.putString("username", emailField);
         editor.apply();
-
     }
 
     public void doLogin(String emailField, String passwordField) {
         //Logica para hacer el login al Back-END
         getView().onLoginSuccess();
-
     }
 
+    public boolean ValidationFields(String emailField, String passwordField) {
+        if (emailField.isEmpty())
+            getView().onEmailFieldEmpty();
```

Comment:
This is OK and it's just a commet, but do you think is there a way to implement this to call getView().showValidationError(case) ? the you don't have to implement a method for each error case. Again, It's just a comment